cmd/src/campaign_progress_printer.go

```diff
@@ -223,3 +240,76 @@ func taskStatusText(ts *campaigns.TaskStatus) (string, error) {
 
 	return statusText, nil
 }
+
+func diffLines(ts *campaigns.TaskStatus) ([]string, error) {
+	lines := []string{}
+
+	if ts.IsCompleted() && ts.ChangesetSpec != nil {
+		fileDiffs, err := diff.ParseMultiFileDiff([]byte(ts.ChangesetSpec.Commits[0].Diff))
+		if err != nil {
+			return lines, err
+		}
+
+		maxFilenameLen := 0
+		fileStats := make(map[string]string, len(fileDiffs))
+
+		for _, f := range fileDiffs {
+			name := f.NewName
+			if name == "/dev/null" {
+				name = f.OrigName
+			}
+
+			if len(name) > maxFilenameLen {
+				maxFilenameLen = len(name)
+			}
+
+			stat := f.Stat()
+			num := stat.Added + 2*stat.Changed + stat.Deleted
+
+			fileStats[name] = fmt.Sprintf("%d %s", num, diffStatDiagram(stat))
+		}
+
+		for file, stats := range fileStats {
+			lines = append(lines, fmt.Sprintf("%-*s | %s", maxFilenameLen, file, stats))
+		}
+	}
+
+	return lines, nil
+}
+
+func finalDiffStatLine(ts *campaigns.TaskStatus) (string, error) {
+	if !ts.IsCompleted() || ts.ChangesetSpec == nil {
+		return "", nil
+	}
+
+	fileDiffs, err := diff.ParseMultiFileDiff([]byte(ts.ChangesetSpec.Commits[0].Diff))
```

Comment:
Can we combine `finalDiffStatLine` and `diffStats` above, so we don't have to parse and iterate twice?

Reply:
Done!